src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -2011,24 +2011,35 @@ private IDisposable ModifyLowerIdIfNeeded(DocumentsOperationContext context, Tab
 
         private static LazyStringValue UnwrapLowerIdIfNeeded(JsonOperationContext context, LazyStringValue lowerId)
         {
-            if (lowerId.Size < ConflictedTombstoneOverhead + 1)
+            if (NeedToUnwrapLowerId(lowerId.Buffer, lowerId.Size) == false)
                 return lowerId;
 
-            if (lowerId[lowerId.Size - ConflictedTombstoneOverhead] == SpecialChars.RecordSeparator)
-            {
-                var size = lowerId.Size - ConflictedTombstoneOverhead;
-                var allocated = context.GetMemory(size + 1); // we need this extra byte to mark that there is no escaping
-                allocated.Address[size] = 0;
+            var size = lowerId.Size - ConflictedTombstoneOverhead;
+            var allocated = context.GetMemory(size + 1); // we need this extra byte to mark that there is no escaping
+            allocated.Address[size] = 0;
+            Memory.Copy(allocated.Address, lowerId.Buffer, size);
+            var lsv = context.AllocateStringValue(null, allocated.Address, size);
+            lsv.AllocatedMemoryData = allocated;
+            return lsv;
+        }
 
-                Memory.Copy(allocated.Address, lowerId.Buffer, size);
-                var lsv = context.AllocateStringValue(null, allocated.Address, size);
-                lsv.AllocatedMemoryData = allocated;
-                return lsv;
-            }
+        protected static int UnwrapLowerIdIfNeeded(byte* lowerId,int size)
```

Comment:
let's rename this to `GetSizeOfTombstoneId`

Reply:
done